src/node/gov/api_version.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/http_query.h"
+#include "ccf/json_handler.h"
+
+#include <string>
+
+namespace ccf::gov::endpoints
+{
+  enum class ApiVersion
+  {
+    preview_v1,
+  };
+
+  static constexpr std::pair<ApiVersion, char const*> api_version_strings[] = {
+    {ApiVersion::preview_v1, "2023-06-01-preview"}};
+
+  static const std::string& supported_suffix()
+  {
+    static std::string ss;
+    if (ss.empty())
+    {
+      ss = "Supported versions are:";
```

Comment:
I suspect a fmt::join(", ") would have been ok, we're probably not going to support dozens of versions at the same time.

Reply:
I think there's 2 suggestions here:
- Separating entries with `, ` rather than `\n   `. I prefer the latter as I think it makes it easier to read + copy a single value, but don't think it makes a huge difference either way
- Using `fmt::join` rather than `ss +=`. I believe `fmt::join` needs a single iterator or iterator-like argument, and not sure how we spell iterator-to-first-element-of-each-pair-in-list in C++, which is why I went for the destructured for-loop

Reply:
I'd missed the second point, this is likely feasible with std::transform and a lambda that extracts the second value, but clearly not worth the extra syntax here.

On the first point, I feel like we've kept most (if not all?) HTTP error messages single line, and resorted to odata details where it seemed beneficial like exception details. I suggest it'd be good to check how other Azure services do that.

Reply:
I've updated this to be a single line, and altered the OData error message to match the [guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#api-versioning):

> [✅](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#versioning-api-version-missing) DO return HTTP 400 with error code "MissingApiVersionParameter" and message "The api-version query parameter (?api-version=) is required for all requests" if client omits the api-version query parameter.

> [✅](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#versioning-api-version-unsupported) DO return HTTP 400 with error code "UnsupportedApiVersionValue" and message "Unsupported api-version '{0}'. The supported api-versions are '{1}'." if client passes an api-version value unrecognized by the service. For the supported api-versions, just list all the stable versions still supported by the service and just the latest public preview version (if any).